commands/kubernetes.go

```diff
@@ -222,14 +250,43 @@ func RunKubernetesUpdate(c *CmdConfig) error {
 		return err
 	}
 
+	if update {
+		tryUpdateKubeconfig(kube, clusterID)
+	}
+
 	return displayClusters(c, *cluster)
 }
 
+func tryUpdateKubeconfig(kube do.KubernetesService, clusterID string) {
+	var (
+		kubeconfig []byte
+		err        error
+	)
+	for tries := 0; ; tries++ {
+		kubeconfig, err = kube.GetKubeConfig(clusterID)
+		if err == nil {
+			break
+		}
+		if tries >= maxAPIFailures {
+			warn(fmt.Sprintf("couldn't get credentials for cluster, it will not be added to your kubeconfig: %v", err))
+			return
+		}
+		time.Sleep(2 * time.Second)
+	}
+	if err := writeOrAddToKubeconfig(kubeconfig); err != nil {
+		warn(fmt.Sprintf("couldn't write cluster credentials: %v", err))
+	}
+}
```

Comment:
Why don't you return an error and print it here instead ?

Reply:
because this isn't a reason to fail the whole operation and not print the cluster that was just created, it would confuse users into thinking their cluster wasn't created